src/WebJobs.Script/Scale/FunctionsScaleManager.cs

```diff
@@ -84,11 +105,52 @@ public virtual async Task<ScaleStatusResult> GetScaleStatusAsync(ScaleStatusCont
                 // this can happen if the host is offline
             }
 
-            var vote = GetAggregateScaleVote(votes, context, _logger);
+            List<int> targetScaleVotes = new List<int>();
+
+            if (targetScalersToProcess.Any())
+            {
+                _logger.LogDebug($"{targetScalersToProcess.Count()} target scalers to sample");
+                HostConcurrencySnapshot snapshot = null;
+                try
+                {
+                    snapshot = await _concurrencyStatusRepository.ReadAsync(CancellationToken.None);
+                }
+                catch (Exception exc) when (!exc.IsFatal())
+                {
+                    _logger.LogError(exc, $"Failed to read concurrency status repository");
```

Comment:
If we have a failure here, is it really the right thing to do to silently fail and continue to compute scale status based on incomplete info?

Reply:
yes, as a `TargetScaler` has default hardcoded concurrency. Also is looks like a exception will be thrown [here ](https://github.com/Azure/azure-webjobs-sdk/blob/dev/src/Microsoft.Azure.WebJobs.Host.Storage/BlobStorageConcurrencyStatusRepository.cs#L61) if `snapshotPersistenceEnabled": true`